internal/docker/docker.go

```diff
@@ -217,7 +218,39 @@ func (c *Docker) Configure(ctx context.Context, request *ConfigureRequest) error
 		fmt.Sprintf("%s:/source:ro", request.ApiRoot), // readonly volume
 	}
 
-	return c.runDocker(ctx, request.Cfg, CommandConfigure, mounts, commandArgs)
+	if err := c.runDocker(ctx, request.Cfg, CommandConfigure, mounts, commandArgs); err != nil {
+		return "", err
+	}
+
+	// Read the new library state from the response.
+	libraryState, err := config.ReadResponse(
+		func(data []byte, libraryState *config.LibraryState) error {
+			return json.Unmarshal(data, libraryState)
+		},
+		filepath.Join(request.RepoDir, config.LibrarianDir, config.ConfigureResponse))
+	if err != nil {
+		return "", err
+	}
+
+	// Update the library state in the librarian state.
+	for i, library := range request.State.Libraries {
+		if library.ID != libraryState.ID {
+			continue
+		}
+		request.State.Libraries[i] = libraryState
+	}
+
+	// Write the updated librarian state to state.yaml.
+	if err := config.WriteLibrarianState(
+		func(state *config.LibrarianState) ([]byte, error) {
+			return yaml.Marshal(state)
+		},
+		request.State,
+		filepath.Join(request.RepoDir, config.LibrarianDir, config.PipelineStateFile)); err != nil {
```

Comment:
This parameter can be just `repodir` instead of `yamlFilePath`, and path constructed inside WriteLibrarianState. This way config.PipelineStateFile does not need to be exported?

Reply:
Done.